python_modules/dagster/dagster_tests/core_tests/definitions_tests/decorators_tests/test_op.py

```diff
@@ -1137,11 +1136,18 @@ def basic() -> List[DynamicOutput]:
     def basic_yield():
         pass
 
-    result = execute_op_in_graph(basic_yield)
-    assert result.success
+    with pytest.raises(
+        DagsterInvariantViolationError,
+        match=r"dynamic output 'result' expected a list of DynamicOutput objects, but instead received instead an object of type \<class 'NoneType'\>\.",
+    ):
+        execute_op_in_graph(basic_yield)
```

Comment:
new error state: previously, an op with a non-optional dynamic output and no return would silently complete. This now results in a failure.